pulsar-client/src/main/java/org/apache/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -59,7 +62,7 @@ public BinaryProtoLookupService(PulsarClientImpl client, String serviceUrl, bool
      * @param destination: topic-name
      * @return broker-socket-address that serves given topic
      */
-    public CompletableFuture<InetSocketAddress> getBroker(DestinationName destination) {
+    public CompletableFuture<Pair<InetSocketAddress, InetSocketAddress>> getBroker(DestinationName destination) {
```

Comment:
In this case it would be the same number of object allocations, either a `Pair<>` or the `InetSocketAddress[]`. No differences, except `Pair` conveys better the fact that 2 addresses are going to be there.